File: frontend/lib/models/situation.dart
class Situation {
  final int id;
  final String description;
  final String language;

  const Situation({
    required this.id,
    required this.description,
    required this.language,
  });

  factory Situation.fromJson(Map<String, dynamic> json, String language) {
    return Situation(
      id: json['id'] as int,
      description: json['description'] as String,
      language: language,
    );
  }
}

class LanguageGroup {
  final String language;
  final List<Situation> situations;

  const LanguageGroup({required this.language, required this.situations});

  factory LanguageGroup.fromJson(Map<String, dynamic> json) {
    final language = json['language'] as String;
    final situations = (json['situations'] as List)
        .map((s) => Situation.fromJson(s as Map<String, dynamic>, language))
        .toList();
    return LanguageGroup(language: language, situations: situations);
  }
}

File: vue/src/entities/situation/situation.ts
import { apiGet } from '@/db/api-client'

export interface Situation {
  id: number
  description: string
  language: string
}

export interface LanguageGroup {
  language: string
  situations: Situation[]
}

interface RawLanguageGroup {
  language: string
  situations: { id: number; description: string }[]
}

function toLanguageGroup(raw: RawLanguageGroup): LanguageGroup {
  return {
    language: raw.language,
    situations: raw.situations.map((s) => ({ ...s, language: raw.language })),
  }
}

export async function fetchGroupedSituations(): Promise<LanguageGroup[]> {
  const data = (await apiGet('/situations/')) as RawLanguageGroup[]
  return data.map(toLanguageGroup)
}

export async function listSituationsForLanguage(languageCode: string): Promise<Situation[]> {
  const data = (await apiGet('/situations/', { language: languageCode })) as RawLanguageGroup[]
  if (data.length === 0) return []
  return toLanguageGroup(data[0]).situations
}
